### PR TITLE
Created project section dynamically and added project details.

### PR DESCRIPTION
**Pull request by batoolfatima2135**
In this branch, I have  :

1. Used a JavaScript array to store all of the information for all projects.
2. Updated the main page so that the projects section is created dynamically using the information stored in that JavaScript object.
3. Implemented the popup window for both mobile and desktop.

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @batoolfatima2135,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that required **Pair-programming** with your coding partners but the Pull request has been performed by one contributor. Make sure your **Partner(s)** has/have some commit contributions before you request another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

**Comment by batoolfatima2135** [comment]
@tresorsawasawa, This is my solo activity. Microverse has allowed me to do all activities solo this week due to internet connection issues in my country. Attaching an email snapshot also for reference
Screenshot 2023-05-17 064944
.

kindly proceed with review

**Comment by tresorsawasawa** [comment]
Hi @batoolfatima2135 

As you have that prof just asked for another review and your project will be reviewed

File: index.html [hunk 1]
```diff
@@ -79,112 +79,9 @@ <h2 class="desktop-view-heading">My Recent Works</h2>
...
             </div>
-
-            <div class="work-item">
-                <div class="snapshot">
-                    <!--img tag should be here but figma template didn't have any image -->
-                </div>
-                <div class="details">
-                    <h2>Multi-Post Stories Gain+Glory</h2>
-                    <ul class="tech-stack">
-                        <li>Ruby on Rails</li>
```
Comment (bobb-Rob) [inline index.html:90]: - [ ] Please see the image below and **note** the portion _circled red_. After you dynamically render these project items, the section title `My recent works` and `See project` buttons _circled red_ on each card are behind the project item card. There should be proper spacing between cards and the section title. Kindly fix this

Your work
image

Figma design
image

File: projects.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  },
+  {
+    title: ' Keeping track of hundreds of components',
+    description1: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s.",
+    description2: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s.",
+    technologies: ['Ruby on Rails', 'CSS', 'Javascript', 'HTML'],
+    image: 'images/work/snapshoot.png',
+    live_version: 'https://batoolfatima2135.github.io/Portfolio/',
+    source: 'https://github.com/batoolfatima2135/Portfolio',
+  },
```
Comment (bobb-Rob) [inline projects.js]: - [ ] [OPTIONAL] Please update each project in the array with unique data. This way it is easy to tell which project was clicked.

File: projects.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    iconsource.className = 'project-icons';
+    iconsource.src = 'images/source.png';
+    demo.appendChild(iconlive);
+    source.appendChild(iconsource);
+    imgdiv.appendChild(img);
+    demo.setAttribute('href', work.live_version);
+    source.setAttribute('href', work.source);
+    Btndivinner.appendChild(demo);
+    Btndivinner.appendChild(source);
+    modal.appendChild(Btndivinner);
```
Comment (bobb-Rob) [inline projects.js:190]: - [ ] Your popup modal design is very different from the provided Figma design. Notice the correct positions of the `close icons`, the two buttons - `see live`, and `see source`, and even the text - title and paragraphs in the Figma design in comparison to your design so far. Please take time and ensure that you stick to the design as much as possible. 
- [ ] Also, please ensure that your project cards and modal popup **TEXT** are using the correct styles - `font-size',  `text-color`, 
- [ ] There should be adequate spaces on the `top`, `right`, `left`, and `bottom` around the modal as shown in the Figma image below. Please fix this

Your design
image

Figma Design
image

File: projects.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    const iconsource = new Image();
+    iconsource.className = 'project-icons';
+    iconsource.src = 'images/source.png';
+    demo.appendChild(iconlive);
+    source.appendChild(iconsource);
+    imgdiv.appendChild(img);
+    demo.setAttribute('href', work.live_version);
+    source.setAttribute('href', work.source);
+    Btndivinner.appendChild(demo);
+    Btndivinner.appendChild(source);
```
Comment (bobb-Rob) [inline projects.js]: - [ ] [OPTIONAL] This dynamic display of project items code is too large and not easy to follow. I suggest that you break down this code into smaller functions and then invoke each function when it should. See code below  👇🏾 

```javascript
function createWorkItem(work) {
  const Workitem = document.createElement('div');
  Workitem.className = 'work-item';
  const imgdiv = createImageDiv(work);
  Workitem.appendChild(imgdiv);

  const detaildiv = createDetailsDiv(work);
  Workitem.append(detaildiv);
  return Workitem;
}

function createImageDiv(work) {
  const imgdiv = document.createElement('div');
  imgdiv.className = 'snapshot';
  const img = new Image();
  img.src = work.image;
  img.className = 'image';
  imgdiv.appendChild(img);
  return imgdiv;
}

function createDetailsDiv(work) {
  const detaildiv = document.createElement('div');
  detaildiv.className = 'details';
  const heading = createHeading(work.title);
  detaildiv.appendChild(heading);

  const ul = createUL(work.technologies);
  detaildiv.appendChild(ul);

  const btndiv = createButtonDiv(work);
  detaildiv.appendChild(btndiv);

  return detaildiv;
}

function createHeading(title) {
  const heading = document.createElement('h2');
  heading.textContent = title;
  return heading;
}

function createUL(technologies) {
  const ul = document.createElement('ul');
  ul.className = 'tech-stack';
  technologies.forEach((value) => {
    const li = document.createElement('li');
    li.textContent = value;
    ul.appendChild(li);
  });
  return ul;
}

function createButtonDiv(work) {
  const btndiv = document.createElement('div');
  btndiv.className = 'project';
  const btn = createButton(work);
  btndiv.appendChild(btn);
  return btndiv;
}

function createButton(work) {
  const btn = document.createElement('button');
  btn.textContent = 'See Project';
  btn.addEventListener('click', () => {
    showModal(work);
  });
  return btn;
}

function showModal(work) {
  const main = createMain(work);
  const modal = createModal(work);
  const cancelicondesktop = createCancelIconDesktop();
  const cancelicon = createCancelIcon();
  cancelicon.addEventListener(
    'click',
    closeProject(main, btndiv, ul)
  );
  cancelicondesktop.addEventListener(
    'click',
    closeProject(main, btndiv, ul)
  );

  const imagediv = createImageDiv(work);
  modal.appendChild(imagediv);
  const h2 = createH2(work.title);
  modal.appendChild(h2);
  const Uldetail = createULDetail(work.technologies);
  modal.appendChild(Uldetail);
  const p = createP(work.description1);
  const p2 = createP(work.description2);
  modal.appendChild(p);
  modal.appendChild(p2);
  const Btndivinner = createButtonDivInner(work);
  modal.appendChild(Btndivinner);

  Workitem.appendChild(imagediv);
  main.append(modal);
  btndiv.appendChild(cancelicondesktop);
  btndiv.appendChild(cancelicon);
  document.body.style.overflow = 'hidden';
}

function createMain() {
  const main = document.createElement('div');
  main.className = 'main-modal';
  Workitem.append(main);
  return main;
}

function createModal() {
  const modal = document.createElement('div');
  modal.className = 'modal';
  return modal;
}

function createCancelIconDesktop() {
  const cancelicondesktop = new Image();
  cancelicondesktop.src = 'images/Disabled.png';
  cancelicondesktop.className = 'cancel-icon-desktop';
  return cancelicondesktop;
}

function createCancelIcon() {
  const cancelicon = new Image();
  cancelicon.src = 'images/cancel.png';
  cancelicon.className = 'cancel-icon';
  return cancelicon;
}

function closeProject(main, btndiv, ul) {
  return () => {
    main.remove();
    btndiv.appendChild(createButton(work));
    detaildiv.appendChild(ul);
    detaildiv.appendChild(btndiv);
    document.body.style.overflow = 'visible';
  };
}

function createH2(title) {
  const h2 = document.createElement('h2');
  h2.className = 'title-project';
  h2.textContent = title;
  return h2;
}

function createULDetail(technologies) {
  const Uldetail = document.createElement('ul');
  Uldetail.className = 'tech-stack-detail';
  technologies.forEach((value) => {
    const Lidetail = document.createElement('li');
    Lidetail.textContent = value;
    Uldetail.appendChild(Lidetail);
  });
  return Uldetail;
}

function createP(description) {
  const p = document.createElement('p');
  p.className = 'description';
  p.textContent = description;
  return p;
}

```

This is not all, but it is enough to give you an example of what it should be.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @batoolfatima2135, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Strangeal — APPROVED** [review]
Hi @batoolfatima2135, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**